swan-lake/development-tutorials/test-document-the-code/test-ballerina-code/execute-tests-in-parallel.md

```diff
@@ -0,0 +1,113 @@
+---
+layout: ballerina-testing-code-left-nav-pages-swanlake
+title: Execute tests in parallel
+description: Learn how to execute tests in parallel.
+keywords: ballerina, programming language, testing, test execution
+permalink: /learn/test-ballerina-code/execute-tests-in-parallel/
+active: execute-tests
+intro: The sections below include information about executing tests in parallel.
+---
+
+## Enable parallel execution of tests
+
+Tests are executed serially by default. User has to consciously enable the parallel execution by passing
+the `--parallel` flag as follows.
+
+```bash
+bal test --parallel
+```
+
+Prior to executing tests concurrently, it is imperative to guarantee concurrency safety. Ballerina inherently assesses the concurrency safety of tests to a certain degree using a predefined set of rules. Tests failing to adhere to these rules are executed serially, notwithstanding the activation of the parallel flag.
+
+## Exclude a specific test from parallel execution
+
+Tests can be flagged to be executed serially eventhough parallel execution is enabled. The common use case is when
+certain tests have dependencies or requirements that make them incompatible with parallel execution. Eg- Shared
+resource/ specific environment set-up. If such a requirment is identfied, User may need to set the `serialExecution`
+flag to `true` as follows.
+
+```ballerina
+@test:Config {serialExecution: true}
+function testAssertEquals6() {
+    test:assertEquals(100, 100);
+}
+```
+
+## Write a concurrent safe test case
+
+Following set of rules should be followed while writing a parallel test,
+
+1) Test function should be isolated. In some instances, compiler infer the test functions as isolated automatically if
```

Comment:
```suggestion
1) Test function should be isolated. In some instances, compiler infers the test functions as isolated automatically if
```

Reply:
"*the* compiler *infers*" sounds more correct for me